### PR TITLE
Update notify_slack action

### PR DESCRIPTION
**Pull request by avidit**
- add flag `include_commit_info`
- use python shell to create payload file
- use payload file to send message

### CODE REVIEW

**Review by QilongTang — APPROVED** [review]

File: notify_slack/action.yml [hunk 1]
```diff
@@ -8,63 +8,84 @@ inputs:
...
+    default: 'false'
+    required: false
 
 runs:
   using: composite
   steps:
-    - name: Set message
-      id: set_message
+    - name: Checkout
+      uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
```
Comment (QilongTang) [inline notify_slack/action.yml:20]: The long numbers always bothers me, is it some kind of SHA of commit?

Comment (avidit) [reply notify_slack/action.yml:20]: yes, it is one of the recommended best practices.
https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

**Comment by QilongTang** [comment]
@avidit The first check failed, expected?

**Comment by avidit** [comment]
> @avidit The first check failed, expected?

Yes because the workflow needs the changes from this pr to be present in the master branch. We need to improve the testing process.

**Pull request merged**